API Blueprint Specification.md

```diff
@@ -478,6 +490,232 @@ Name and description of the API
+
+This section **must** be composed of nested list items only. This section **must not** contain any other elements. Each list item describes a single Authentication scheme.
+
+    + <scheme name> (<base scheme type>) - <description>
+
+        + <customizable option 1>
+        + <customizable option 2>
+        ...
+        + <customizable option N>
+
+Where:
+
++ `<scheme name>` is the authentication scheme name by which this scheme is represented in [Authenticated section](#def-authenticated-section).
++ `<description>` is any **optional** Markdown-formatted description of the authentication scheme.
++ `<base scheme type>` is the base authentication scheme type (e.g. "Basic Authentication Scheme"). See below for all the available base authentication schemes.
++ `<customizable option n>` represents a customizable option in the base authentication scheme. It is a [MSON Property Member Declaration][] without the [MSON Type Definition][] where the [MSON Property Name][] represents the name of the option and [MSON Value][] represents the value of the option.
+
+List of the base authentication schemes supported by API Blueprint are described below.
+
+#### Basic Authentication Scheme
+This is the [Basic Authentication Scheme](http://tools.ietf.org/html/rfc2617#section-2) as specified in [RFC 2617](http://tools.ietf.org/html/rfc2617).
+
+The customizable options for this scheme are:
+
++ `example_username` represents an example username value. Value type is **string**. "example_user" is the **default**.
++ `example_password` represents an example password value. Value type is **string**. "example_pass" is the **default**.
+
+##### Example
+
+    + basic (Basic Authentication Scheme)
+        + example_username: `john`
+        + example_password: `password`
+
+#### OAuth2 Authentication Scheme
+This scheme uses the [OAuth2 Authorization Framework](http://tools.ietf.org/html/rfc6749) specified in [RFC 6749](http://tools.ietf.org/html/rfc6749) to get an authorization grant which is then used to access the API.
+
+The customizable options for this scheme are:
+
++ `oauth_resource_endpoint` is the oauth resource endpoint where the oauth framework resides. Value type is **string**. "/" is the **default**.
++ `scopes` represents the OAuth scopes which indicate a list of permissions. Value type is **array**. By **default** no scopes are specified.
++ `authorization_endpoint` is the authorization endpoint required for the authorization grant. Value type is **string**. "/authorize" is the **default**.
++ `access_token_endpoint` is the token endpoint used by client to obtain the access token. Value type is **string**. "/access_token" is the **default**.
++ `revocation_endpoint` is the revocation endpoint used by client to revoke a token. Value type is **string**. "/revoke" is the **default**.
```

Comment:
I wasn't aware of RFC7009, since this isn't actually part of the original RFC for OAuth2 (RFC6746) and instead added as a separate RFC a year later. There will be many implementations that don't offer this, or offer it in a different way. As such I still believe we should not have a default value and make this opt-in for the many APIs that do not support RFC7009.

For example, GitHub offer this in a different way in their OAuth2 implementation where they use DELETE. https://developer.github.com/v3/oauth_authorizations/#revoke-an-authorization-for-an-application